hazelcast/src/main/java/com/hazelcast/query/impl/BaseSingleValueIndexStore.java

```diff
@@ -27,13 +27,24 @@
  * The base store for indexes that are unable to work with multi-value
  * attributes natively. For such indexes {@link MultiResult}s are split into
  * individual values and each value is inserted/removed separately.
+ * <p>
+ * All operations on the index store like insert/remove/getRecords are
+ * not guarded by a global lock. The subclasses must either implement thread-safe
```

Comment:
The javadoc looks outdated comparing to the actual code.

Reply:
Fixed